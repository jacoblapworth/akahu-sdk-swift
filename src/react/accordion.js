import XUIAccordion from './components/accordion/XUIAccordion';

export default XUIAccordion;
