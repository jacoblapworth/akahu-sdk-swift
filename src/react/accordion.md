<div class="xui-margin-vertical">
		<a href="../section-compounds-displayingdata-accordion.html" isDocLink>Accordion in the XUI Documentation</a>
</div>

Accordions are used to display a vertically expandable & collapsible list that reveals and hides additional content

The _Accordion_ is broken up into two key components `import XUIAccordion, { XUIAccordionItem } from '@xero/xui/react/accordion';`.



## Basic

The bare minimum _Accordion_ composition can be achieved with the `items` and `createItem` prop.

+ `items` represents the generic data structure of each _Accordion_ item.
+ `createItem` returns a `<XUIAccordionItem />` that represents a single _item_ from the `items` prop.

**Note:** Each item needs a unique `id`. By default an object _key_ named `id` is looked up in each item however you can specify a new key if there is a more relevant unique identifier using the prop `idKey`.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
<XUIAccordion
	items={[
		{ id: 1, name: 'John Smith', content: 'Accountant' },
		{ id: 2, name: 'Barry Allen', content: 'Bookkeeper' },
		{ id: 3, name: 'Ernest Hemmingway' }
	]}
	createItem={({ name, content }) => (
		<XUIAccordionItem
			primaryHeading={name}
			onItemClick={({id, isOpen}) => console.log(`clicked! ${id} | ${isOpen}`)}>
			{ content &&  <div className="xui-padding-horizontal-large xui-padding-vertical-4xlarge">{content}</div> }
		</XUIAccordionItem>
	)}
/>
```

## Accordion Item

The `<XUIAccordionItem />` can take a variety of props that can inject content into the layout (`primaryHeading`, `secondaryHeading`, `leftContent`, `pinnedValue`, `action`, `overflow`, `custom`).

You can also supply an `onItemClick` prop that returns the entire item from the items array in addition to an `isOpen` boolean representing the items toggled state.

**Note:** if you are adding a `<button />` type element as content remember to [`stopPropagation`](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation) so the event does not bubble up and trigger the _Accordion_ _expand_ / _collapse_ toggle accidentally.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const itemStyle = { background: 'lightgray', border: '1px solid darkgray', padding: '4px', outline: '1px solid gray' };
<XUIAccordion
	items={[{ id: 1 }]}
	createItem={() => (
		<XUIAccordionItem
			primaryHeading={<div style={itemStyle}>Primary Heading</div>}
			secondaryHeading={<div style={itemStyle}>Secondary Heading</div>}
			leftContent={<div style={itemStyle}>Left Content</div>}
			pinnedValue={<div style={itemStyle}>Pinned Value</div>}
			action={<div style={itemStyle}>Action</div>}
			overflow={<div style={itemStyle}>Overflow</div>}
			custom={<div style={itemStyle}>Custom Content</div>}
			onItemClick={({isOpen}) => alert(`${isOpen ? 'Open' : 'Close'} example item`)}
		/>
	)}
/>
```

## Empty State

If no `children` content is supplied to a `<XUIAccordionItem />` then the _empty state_ component will be rendered instead.

There is a default design that can be overridden or completely replaced (`emptyMessage`, `emptyIcon`, `emptyStateComponent`).

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const starIconPath = require ('@xero/xui-icon/icons/star').default;
const XUIAvatar = require('./avatar').default;
const XUIButton = require('./button').default;
const XUIIcon = require('./icon').default;
const icon = require('@xero/xui-icon/icons/overflow').default;
<div>
	<XUIAccordion
		className="xui-margin-bottom-large"
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Default empty state" />}
	/>
	<XUIAccordion
		className="xui-margin-bottom-large"
		emptyIcon={starIconPath}
		emptyMessage="Custom empty state message"
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Custom empty state" />}
	/>
	<XUIAccordion
		emptyStateComponent={(
			<div
				className="xui-heading xui-textcolor-inverted xui-padding-vertical-large xui-text-align-center"
				style={ { background: 'darkgray' } }>
				Replace empty state component
			</div>
		)}
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Replace empty state" />}
	/>
</div>
```

## Example

You can nest other XUI component inside the `<XUIAccordionItem />` _(such as `<XUIContentBlock />`)_ to create robust compositions.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const { XUIContentBlock, XUIContentBlockItem } = require('./structural.js');
const { default: XUIAvatar } = require('./avatar.js');
const { default: XUIButton } = require('./button.js');
const { default: XUIIcon } = require('./icon.js');
const { default: overflowPathData } = require('@xero/xui-icon/icons/overflow');
<XUIAccordion
	idKey="name"
	items={[
		{
			name: 'John Smith',
			contacts: [
				{ contact: 'Peggy Olsen', minutes: '00:20' },
				{ contact: 'Pete Campbell', minutes: '15:30' }
			]
		}
	]}
	createItem={({ name, contacts }) => (
		<XUIAccordionItem
			primaryHeading={name}
			leftContent={<XUIAvatar value={name} className="xui-margin-right"/>}
			overflow={<XUIButton variant="icon"><XUIIcon icon={overflowPathData} title="Overflow menu"/></XUIButton>}
			action={(
				<XUIButton
					size="small"
					className="xui-margin-right-small"
					onClick={event => {
						event.stopPropagation();
						alert('Clicked update button');
					}}>Update</XUIButton>
			)}>
				<XUIContentBlock>
					{ contacts.map(({contact, minutes}) => (
						<XUIContentBlockItem
							key={contact}
							primaryHeading={contact}
							pinnedValue={minutes}
							href="#"
						/>
					)) }
				</XUIContentBlock>
		</XUIAccordionItem>
	)}
/>
```
