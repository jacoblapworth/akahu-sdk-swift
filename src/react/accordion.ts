import XUIAccordion from './components/accordion/XUIAccordion';
import XUIAccordionItem from './components/accordion/XUIAccordionItem';

export { XUIAccordion as default, XUIAccordion, XUIAccordionItem };
