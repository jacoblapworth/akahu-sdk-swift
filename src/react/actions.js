import XUIActions from './components/actions/XUIActions';

export default XUIActions;
