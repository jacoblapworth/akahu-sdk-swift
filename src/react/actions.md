<div class="xui-margin-vertical">
	<a href="../section-components-controls-button.html#components-controls-button-12" isDocLink>Actions in the XUI Documentation</a>
</div>

Actions are used to wrap a selection of buttons (often two) and provide formatting. They can be embedded in a XUIModal, in a XUIPageHeader, in the footer of a XUIPanel, and other places. These are more generic than Banner Actions or Toast Actions.

#### Default action

A default action aligns a primary and secondary button to the right, with the primary as the rightmost.

```jsx harmony
import XUIActions from '@xero/xui/react/actions';
import XUIButton from '@xero/xui/react/button';

<XUIActions
  primaryAction={<XUIButton variant="main">Continue</XUIButton>}
  secondaryAction={<XUIButton>Cancel</XUIButton>}
/>;
```

#### Linear action

The linear variant aligns the primary button to the right and the secondary button to the left.

```jsx harmony
import XUIActions from '@xero/xui/react/actions';
import XUIButton from '@xero/xui/react/button';

<XUIActions
  isLinear
  primaryAction={<XUIButton variant="main">Next</XUIButton>}
  secondaryAction={<XUIButton>Previous</XUIButton>}
/>;
```
