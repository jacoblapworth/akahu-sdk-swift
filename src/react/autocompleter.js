import Autocompleter from './components/autocompleter/Autocompleter';
import AutocompleterInput from './components/autocompleter/AutocompleterInput';
import SecondarySearch from './components/autocompleter/SecondarySearch';
import EmptyState from './components/autocompleter/EmptyState';
import { boldMatch, decorateSubStr } from './components/autocompleter/helpers/highlighting';

export {
	Autocompleter as default,
	AutocompleterInput,
	EmptyState,
	SecondarySearch,
	boldMatch,
	decorateSubStr
};
