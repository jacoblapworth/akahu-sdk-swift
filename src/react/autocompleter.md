<div class="xui-margin-vertical">
	<a isDocLink href="../section-compounds-collectinginput-autocompleter.html">Autocompleter in the XUI Documentation</a>
</div>
`XUIAutocompleter` is a component that composes many other components together. It's an input where users can type to filter a list of items to select.

Refer to the following sections of the XUI Documentation for more information about the components that make up a `XUIAutocompleter`.

##### Related Components

<div class="xui-margin-vertical">
	<div>
		<a href="#text-input" isDocLink>Text Input</a>
	</div>
	<div>
		<a href="#avatar" isDocLink>Avatar</a>
	</div>
	<div>
		<a href="#dropdown" isDocLink>Dropdown</a>
	</div>
	<div>
		<a href="#pill" isDocLink>Pill</a>
	</div>
	<div>
		<a href="#tag" isDocLink>Tag</a>
	</div>
</div>

## Examples

### Multi-select

`XUIAutocompleter` can be passed an array of pills to display to the left of the input. Each pill should have the `xui-autocompleter--pill` class applied to receive the correct padding. `XUIAutocompleter` also provides a configurable empty state for when no search results are returned.

You should add a callback to `onBackspacePill` which removes the last selected element. This will be called if the backspace key is pressed while the input is empty.

Also note that the `<XUIAutocompleterEmptyState>` component needs to be wrapped in a `<Picklist>` component if you want to be able to access a header or footer with the keyboard.

Currently it's recommended that you do not pass in a `rightElement` prop while using wrapping pills. At certain widths, the right element may wrap down to the next row without the input, which doesn't display nicely.

```jsx
const { boldMatch, decorateSubStr } = require('./autocompleter');
const XUIAutocompleterEmptyState = require('./components/autocompleter/XUIAutocompleterEmptyState').default;
const { Component } = require('react');
const people  = require('./components/autocompleter/private/people').default;
const Pickitem = require('./components/picklist/Pickitem').default;

const filterPeople = (peopleToSearch, value, idsToExclude) => {
	const val = value.toLowerCase();
	return peopleToSearch.filter((person, index) =>
		idsToExclude.indexOf(index) === -1 &&
		(person.name.toLowerCase().indexOf(val) > -1
		|| person.email.toLowerCase().indexOf(val) > -1
		|| person.subtext.toLowerCase().indexOf(val) > -1)
	);
};

//Example to show how the children can be styled however and you also define your own search criteria.
class WrapPillsExample extends Component {
	constructor() {
		super();

		this.state = {
			value: '',
			selectedPeopleIds: [0]
		};

		this.onSearchChangeHandler = this.onSearchChangeHandler.bind(this);
		this.deletePerson = this.deletePerson.bind(this);
		this.deleteLastPerson = this.deleteLastPerson.bind(this);
		this.selectPerson = this.selectPerson.bind(this);
	}

	onSearchChangeHandler(value) {
		this.completer.openDropDown();
		this.setState(prevState => ({ value }));
	}

	deletePerson(idToRemove) {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds.filter(id => id !== idToRemove)],
		}));
	}

	deleteLastPerson() {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds.slice(0, -1)],
		}));
	}

	selectPerson(person) {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds, person],
			value: '',
		}));
	}

	render(){
		const { value, selectedPeopleIds } = this.state;
		const unselectedPeopleIds = filterPeople(people, value, selectedPeopleIds);

		const dropdownContents = unselectedPeopleIds.length === 0 ?
			(
				<Picklist>
					<XUIAutocompleterEmptyState id="no_people">
						No People Found
					</XUIAutocompleterEmptyState>
				</Picklist>
			) : (
				<Picklist>
					{unselectedPeopleIds.map(person => (
						<Pickitem key={person.id} id={person.id} value={person.id} onSelect={this.selectPerson}>
							<div className="xui-u-flex">
								<XUIAvatar value={person.name} imageUrl={person.avatar} />
								<div className="xui-u-grow xui-padding-left">
									<div className="xui-heading-item xui-text-truncated">
										{decorateSubStr(person.name, value || '', boldMatch)}
									</div>
									<div className="xui-text-secondary xui-text-truncated">
										{decorateSubStr(person.email, value || '', boldMatch)}, {decorateSubStr(person.subtext, value || '', boldMatch)}
									</div>
								</div>
							</div>
						</Pickitem>
					))}
				</Picklist>
			);

		return (
				<XUIAutocompleter
					inputLabelText='autocompleter'
					isInputLabelHidden
					ref={ac => this.completer = ac}
					onSearch={this.onSearchChangeHandler}
					placeholder="XUI Autocompleter accommodates enough space to fit the placeholder"
					searchValue={value}
					onBackspacePill={this.deleteLastPerson}
					pills={
						selectedPeopleIds.map(id =>
							<XUIPill
								value={people[id].name}
								className="xui-autocompleter--pill"
								onDeleteClick={()=>this.deletePerson(id)}
								isMaxContentWidth
								key={id}
							/>
						)
					}
				>
					{dropdownContents}
				</XUIAutocompleter>
		)
	}
}

<WrapPillsExample />
```

### Multi-select without wrapping pills

By default the pills and search bar will wrap inside the `XUIAutocompleter` input container. To disable this, set `disableWrapPills` to true.

```jsx
const { boldMatch, decorateSubStr } = require('./autocompleter');
const XUIAutocompleterEmptyState = require('./components/autocompleter/XUIAutocompleterEmptyState').default;
const { Component } = require('react');
const people  = require('./components/autocompleter/private/people').default;
const Pickitem = require('./components/picklist/Pickitem').default;

const filterPeople = (peopleToSearch, value, idsToExclude) => {
	const val = value.toLowerCase();
	return peopleToSearch.filter((person, index) =>
		idsToExclude.indexOf(index) === -1 &&
		(person.name.toLowerCase().indexOf(val) > -1
		|| person.email.toLowerCase().indexOf(val) > -1
		|| person.subtext.toLowerCase().indexOf(val) > -1)
	);
};

//Example to show how the children can be styled however and you also define your own search criteria.
class DisableWrapPills extends Component {
	constructor() {
		super();

		this.state = {
			value: '',
			selectedPeopleIds: [0]
		};

		this.onSearchChangeHandler = this.onSearchChangeHandler.bind(this);
		this.deletePerson = this.deletePerson.bind(this);
		this.deleteLastPerson = this.deleteLastPerson.bind(this);
		this.selectPerson = this.selectPerson.bind(this);
	}

	onSearchChangeHandler(value) {
		this.completer.openDropDown();
		this.setState(prevState => ({ value }));
	}

	deletePerson(idToRemove) {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds.filter(id => id !== idToRemove)],
		}));
	}

	deleteLastPerson() {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds.slice(0, -1)],
		}));
	}

	selectPerson(person) {
		this.setState(prevState => ({
			selectedPeopleIds: [...prevState.selectedPeopleIds, person],
			value: '',
		}));
	}

	render(){
		const { value, selectedPeopleIds } = this.state;
		const unselectedPeopleIds = filterPeople(people, value, selectedPeopleIds);

		const dropdownContents = unselectedPeopleIds.length === 0 ?
			(
				<Picklist>
					<XUIAutocompleterEmptyState id="no_people">
						No People Found
					</XUIAutocompleterEmptyState>
				</Picklist>
			) : (
				<Picklist>
					{unselectedPeopleIds.map(person => (
						<Pickitem key={person.id} id={person.id} value={person.id} onSelect={this.selectPerson}>
							<div className="xui-u-flex">
								<XUIAvatar value={person.name} imageUrl={person.avatar} />
								<div className="xui-u-grow xui-padding-left">
									<div className="xui-heading-item xui-text-truncated">
										{decorateSubStr(person.name, value || '', boldMatch)}
									</div>
									<div className="xui-text-secondary xui-text-truncated">
										{decorateSubStr(person.email, value || '', boldMatch)}, {decorateSubStr(person.subtext, value || '', boldMatch)}
									</div>
								</div>
							</div>
						</Pickitem>
					))}
				</Picklist>
			);

		return (
				<XUIAutocompleter
					inputLabelText='autocompleter'
					isInputLabelHidden
					ref={ac => this.completer = ac}
					onSearch={this.onSearchChangeHandler}
					placeholder="XUI Autocompleter accommodates enough space to fit the placeholder"
					searchValue={value}
					onBackspacePill={this.deleteLastPerson}
					disableWrapPills
					pills={
						selectedPeopleIds.map(id =>
							<XUIPill
								value={people[id].name}
								className="xui-autocompleter--pill"
								onDeleteClick={()=>this.deletePerson(id)}
								isMaxContentWidth
								key={id}
							/>
						)
					}
				>
					{dropdownContents}
				</XUIAutocompleter>
		)
	}
}

<DisableWrapPills />
```

### Single-select

When using `XUIAutocompleter` for selecting a single option, use the `leftElement` and `rightElement` props for adding information and options about the selected item and leave the `pills` prop empty.

```jsx
const { boldMatch, decorateSubStr } = require('./autocompleter');
const XUIIcon = require('./components/icon/XUIIcon').default;
const XUIAutocompleterEmptyState = require('./components/autocompleter/XUIAutocompleterEmptyState').default;
const XUITextInputSideElement = require('./components/textInput/XUITextInputSideElement').default;
const { Component } = require('react');
const people  = require('./components/autocompleter/private/people').default;
const Pickitem = require('./components/picklist/Pickitem').default;
const crossIcon = require('@xero/xui-icon/icons/cross-small').default;

const filterPeople = (peopleToSearch, value) => {
	const val = value.toLowerCase();
	return peopleToSearch.filter(person =>
		person.name.toLowerCase().indexOf(val) > -1
		|| person.email.toLowerCase().indexOf(val) > -1
		|| person.subtext.toLowerCase().indexOf(val) > -1
	);
};

//Example to show how the children can be styled however and you also define your own search criteria.
class SingleSelectExample extends Component {
	constructor() {
		super();

		this.state = {
			value: people[0].name,
			selectedPersonId: 0
		};

		this.onSearchChangeHandler = this.onSearchChangeHandler.bind(this);
		this.selectPerson = this.selectPerson.bind(this);
	}

	onSearchChangeHandler(value) {
		this.completer.openDropDown();
		this.setState(prevState => {
			const { selectedPersonId } = prevState;
			const textIsCurrentName = selectedPersonId != null && value === people[selectedPersonId].name;
			return {
				value,
				selectedPersonId: textIsCurrentName ? selectedPersonId : null,
			};
		});
	}

	selectPerson(selectedPersonId) {
		this.setState(prevState => ({
			selectedPersonId,
			value: selectedPersonId != null ? people[selectedPersonId].name : '',
		}));
	}

	render(){
		const { value, selectedPersonId } = this.state;
		const searchResults = filterPeople(people, value);

		const dropdownContents = (
			<Picklist>
				{searchResults.map(person => (
					<Pickitem key={person.id} id={person.id} value={person.id} onSelect={this.selectPerson}>
						<div className="xui-u-flex">
							<XUIAvatar value={person.name} imageUrl={person.avatar} />
							<div className="xui-u-grow xui-padding-left">
								<div className="xui-heading-item xui-text-truncated">
									{decorateSubStr(person.name, value || '', boldMatch)}
								</div>
								<div className="xui-text-secondary xui-text-truncated">
									{decorateSubStr(person.email, value || '', boldMatch)}, {decorateSubStr(person.subtext, value || '', boldMatch)}
								</div>
							</div>
						</div>
					</Pickitem>
				))}
			</Picklist>
		);
		const leftElement = selectedPersonId != null && (
			<XUITextInputSideElement type="avatar" >
				<XUIAvatar
					value={people[selectedPersonId].name}
					imageUrl={people[selectedPersonId].avatar}
					size="small"
				/>
			</XUITextInputSideElement>
		);
		const rightElement = selectedPersonId != null && (
			<XUITextInputSideElement type="icon">
				<XUIButton variant="icon" onClick={() => this.selectPerson(null)} aria-label="Clear">
					<XUIIcon icon={crossIcon} />
				</XUIButton>
			</XUITextInputSideElement>
		);

		return (
			<XUIAutocompleter
				inputLabelText='autocompleter'
				isInputLabelHidden
				ref={ac => this.completer = ac}
				onSearch={this.onSearchChangeHandler}
				placeholder="Select a person"
				searchValue={value}
				onBackspacePill={this.deleteLastPerson}
				leftElement={leftElement}
				rightElement={rightElement}
			>
				{dropdownContents}
			</XUIAutocompleter>
		)
	}
}

<SingleSelectExample />
```
