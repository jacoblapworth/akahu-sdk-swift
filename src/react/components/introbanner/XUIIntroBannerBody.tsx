import cn from 'classnames';
import * as PropTypes from 'prop-types';
import * as React from 'react';

import baseClass from './private/constants';

interface Props {
  /**
   * The text content of the intro banner. We recommend providing this in the form of `<p>` elements.
   */
  children: React.ReactNode;
  className?: string;
  qaHook?: string;
}

const XUIIntroBannerBody = ({ children, className, qaHook, ...spreadProps }: Props) => {
  const classes = cn(`${baseClass}--body`, className);

  return (
    <div {...spreadProps} className={classes} data-automationid={qaHook}>
      {children}
    </div>
  );
};

XUIIntroBannerBody.propTypes = {
  /**
   * The text content of the intro banner. We recommend providing this in the form of `<p>` elements.
   */
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  qaHook: PropTypes.string,
};

export default XUIIntroBannerBody;
