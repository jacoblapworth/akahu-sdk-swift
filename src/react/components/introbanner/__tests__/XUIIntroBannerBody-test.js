import React from 'react';
import { axe, toHaveNoViolations } from 'jest-axe';
import XUIIntroBannerBody from '../XUIIntroBannerBody';

import { render } from '@testing-library/react';

expect.extend(toHaveNoViolations);

const createComponent = props => (
  <XUIIntroBannerBody {...props}>
    <p>Intro Banner Body Text</p>
  </XUIIntroBannerBody>
);

describe('XUIIntroBannerBody', () => {
  test('renders basic example correctly', () => {
    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders a passed qaHook as an automationId', () => {
    const { asFragment } = render(createComponent({ qaHook: 'introbanner-body' }));
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders extra classes when passed to the className prop', () => {
    const { asFragment } = render(createComponent({ className: 'introbanner-body-class' }));
    expect(asFragment()).toMatchSnapshot();
  });

  test('should pass accessibility testing', async () => {
    const { container } = render(createComponent());
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
