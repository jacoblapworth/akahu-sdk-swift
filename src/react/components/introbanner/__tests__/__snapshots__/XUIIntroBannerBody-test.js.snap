// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XUIIntroBannerBody renders a passed qaHook as an automationId 1`] = `
<DocumentFragment>
  <div
    class="xui-introbanner--body"
    data-automationid="introbanner-body"
  >
    <p>
      Intro Banner Body Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`XUIIntroBannerBody renders basic example correctly 1`] = `
<DocumentFragment>
  <div
    class="xui-introbanner--body"
  >
    <p>
      Intro Banner Body Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`XUIIntroBannerBody renders extra classes when passed to the className prop 1`] = `
<DocumentFragment>
  <div
    class="xui-introbanner--body introbanner-body-class"
  >
    <p>
      Intro Banner Body Text
    </p>
  </div>
</DocumentFragment>
`;
